Source/Csla/Core/MobileObservableCollection.cs

```diff
@@ -136,7 +136,7 @@ protected virtual void OnGetChildren(SerializationInfo info, MobileFormatter for
       List<int> references = new List<int>();
       for (int x = 0; x < Count; x++)
       {
-        T? child = this[x];
+        T child = this[x];
```

Comment:
Shouldn`t this still be `T?` because of the next line? Otherwise the check could be removed due to the missing annotation?

Reply:
no because ` MobileObservableCollection<T>` has no `notnull` constraint.

Reply:
But exactly because of that we have to assume it can be nullable.
So `T? child` is more correct than `T`. With `T` we don't express the possibility of the `child` being `null` so the compiler can help us avoiding possible `null` errors?